### ISSUE TITLE
Dralbase TODOs

### ISSUE THREAD

**Issue by fonsp** [fonsp/Pluto.jl#812]
Must haves for Pluto.14
- [x] Run Firebasey tests on CI 🏝
- [x] Make sure /samples/test1.jl runs without issue 🏝
- [x] Notebook request without a notebook error
- [x] Can't assign to read-only notebook error 🏝
- [x] Set cell to 'running' by default (default is weird now) - get overrided by the backend 🗿
- [x] TODOs in code 🏝🗿
- [x] Delete old tests 🗿
- [x] Comments in `src/webserver/Dynamic.jl` 🗿
- [x] Lag issue / type in textboxes due to bond synchronization 🗿
- [x] Can't drag cells from notebook to another
- [x] Verify that the cell memo is actually working 🏝
- [x] Big notebook with images seems slow -- sample here: https://github.com/fonsp/disorganised-mess/blob/master/big_images.jl This notebook became considerably slower 🗿
- [x] Clean up Firebasey.jl 🗿
- [x] Pasting cells using Ctrl+V shows empty cells for a short while? 🏝
- [x] File picker input is broken when multiple browser windows are open -- all other windows crash
- [ ] Do performance timings on a big notebook and slow computer

---

https://github.com/fonsp/Pluto.jl/pull/710

- [x] Fix existing frontend tests
- [x] Bonds keep going for too long?
- [x] Run Firebasey notebook tests on CI
- [x] Rename Firebase to something else like Dralbase or Firebasey
- [x] Rename `cell_inputs`, `cell_inputs_local` & `cell_results` to `cell_dict`, `cells_local` & `cells_running`? Also look at `CellData`, `CellState` (sound like synonyms) and the structs inside Dynamic.jl
- [x] Remove logging code from Dynamic.jl
- [x] Delete old tests from test/Dynamic.jl
- [x] DRAL also wrote some TODOs in code
- [x] Go through the /samples/test1.jl notebook
- [x] Remove the JS debugging thing where we replace the document.body with the error message
- [x] Fix tables

Comments needed for:
- [x] This needs lots of comments: https://github.com/fonsp/Pluto.jl/blob/d3e316ef61382c8c69cf8ea04b30a6a523606b95/src/webserver/Dynamic.jl#L163-L227
- [x] We need a description somewhere in the code to explain how our state management works, a contributor needs to know what to do to add something like Bret Victor logging. Maybe we start by copy pasting the description that I wrote for this PR?

**Comment by fonsp** [comment]
begin-wrapping is broken

**Comment by fonsp** [comment]
<img width="460" alt="image" src="https://user-images.githubusercontent.com/6933510/103323061-6dc6b200-4a41-11eb-958d-bcb97d558809.png">

reevaluate first cell:

<img width="470" alt="image" src="https://user-images.githubusercontent.com/6933510/103323074-7f0fbe80-4a41-11eb-8d6b-52326225ebb4.png">

**Comment by fonsp** [comment]
Rerunning a cell should rerender the HTML, we can use persist_js_state for this

![Schermopname 2020-12-30 om 23 06 29](https://user-images.githubusercontent.com/6933510/103383656-d4a4a380-4af3-11eb-85ad-c0bc90c9d10c.gif)

**Comment by fonsp** [comment]
PlutoUI.MultiSelect is broken

**Comment by fonsp** [comment]
(for myself) Sometimes when I go from one page to another I get this:

![image](https://user-images.githubusercontent.com/6933510/103904910-353c6d80-50fe-11eb-8c9b-f2e128e1732c.png)

But it's fine to ignore

**Comment by fonsp** [comment]
The rectangle drawing widget in the Interactivity sample notebook doesn't work

**Comment by pankgeorg** [comment]
> (for myself) Sometimes when I go from one page to another I get this:
> 
> ![image](https://user-images.githubusercontent.com/6933510/103904910-353c6d80-50fe-11eb-8c9b-f2e128e1732c.png)
> 
> But it's fine to ignore

I often (1,2 times per hour) get this error and an alert about immutable notebook state or something.

![image](https://user-images.githubusercontent.com/8681967/104129889-e4837980-5376-11eb-816a-eb825821db40.png)
![image](https://user-images.githubusercontent.com/8681967/104129912-f238ff00-5376-11eb-878c-a7a4d3bf9369.png)

**Comment by fonsp** [comment]
If there is too much lag between the client and server, you can't type in textboxes:

![Schermopname 2021-01-12 om 14 43 15](https://user-images.githubusercontent.com/6933510/104322564-65f91a00-54e5-11eb-8963-f28c0284e1e2.gif)

**Comment by fonsp** [comment]
File picker input is broken when multiple browser windows are open

**Comment by fonsp** [comment]
The first time you use Pluto and run a cell, it takes very long until the cell starts running, and it looks like it's already done but without output

**Comment by pankgeorg** [comment]
[This](https://github.com/immerjs/immer/issues/576) seems to be related to the readonly property issue, so I'm disabling the auto-freeze at the immer.js import.
```
pgeorgakopoulos@rosetta:~/pluto/Pluto.jl$ git commit -m "Disable immer autofreeze"
[release-14.0.0 93efd83] Disable immer autofreeze
 1 file changed, 2 insertions(+), 1 deletion(-)
```

**Comment by fonsp** [comment]
This notebook is much slower than on 0.12: https://github.com/fonsp/disorganised-mess/blob/master/big_images.jl (try the slider).

It's probably not because of the frontend, I filtered out the last cell in the Notebook component with no difference.

The server uses lots of memory: starting the notebook and refreshing the page a couple of times leads to 3GB ram usage for the server process, calling GC on process 1 did not clear it.

**Comment by fonsp** [comment]
Ctrl+A -> click and drag the text

![Schermopname 2021-02-08 om 17 27 52](https://user-images.githubusercontent.com/6933510/107249248-77c1d480-6a33-11eb-8d5b-c39cf268aa91.gif)

**Comment by pankgeorg** [comment]
> Ctrl+A -> click and drag the text
> 
> ![Schermopname 2021-02-08 om 17 27 52](https://user-images.githubusercontent.com/6933510/107249248-77c1d480-6a33-11eb-8d5b-c39cf268aa91.gif)

on it!

**Comment by fonsp** [comment]
"Peeking into a cell" no longer works.

Say I have two cells, 1 is open, and 2 is folded.
If I am in cell 1, and use the down arrow to navigate into cell 2, it should open (works), but when i navigate away (up arrow), it should fold again (does not work).

**Comment by fonsp** [comment]
https://github.com/fonsp/Pluto.jl/issues/928

**Comment by fonsp** [comment]
https://github.com/fonsp/Pluto.jl/issues/628

**Comment by pankgeorg** [comment]
> "Peeking into a cell" no longer works.
> 
> Say I have two cells, 1 is open, and 2 is folded.
> If I am in cell 1, and use the down arrow to navigate into cell 2, it should open (works), but when i navigate away (up arrow), it should fold again (does not work).

this is also fixed in #628 's fix

**Comment by fonsp** [comment]
Note to self: My update throttling should blablba
